### PR TITLE
API: Added upload functionality (imgs, resrcs); CLIENT: WYSIWYG uploa…

### PR DESCRIPTION
**Pull request by drserg**
…ding image

### CODE REVIEW

File: app/modules/module/views/CreateModuleView.js [hunk 1]
```diff
@@ -22,10 +22,23 @@ define(function(require, exports, module) {
...
-                });
+                _.delay(function() {
+                    var editor = $("#description").ckeditor({
+                            extraPlugins: 'justify,image,uploadimage',
+                            imageUploadUrl: CMS.api+"upload/image",
+                            language: 'uk',
+                            skin:'moono'
+                        }).editor;
+                    editor.on('fileUploadRequest', function( evt ) {
+                            console.log("asdasdasdasd");
```
Comment (okrasn) [inline app/modules/module/views/CreateModuleView.js]: Clear console log please

File: server/controllers/authenticate.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ var express = require('express'),
     Profile = require('../models/profile');
 
 router.post('/', function(req, res) {
-
+    console.log(req.body);
```
Comment (okrasn) [inline server/controllers/authenticate.js]: Delete console.log if not needed

File: server/controllers/uploads.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+router.post('/image', function(req, res) {
+    req.pipe(req.busboy);
+    req.busboy.on('file', function(fieldname, file, filename) {
+        var fstream = fs.createWriteStream('./server/uploads/images/' + filename); 
+        file.pipe(fstream);
+        fstream.on('close', function () {
+            res.json({
+                "uploaded": 1,
+                "fileName": filename,
+                "url": "http://localhost:8888/uploads/images/"+filename
```
Comment (okrasn) [inline server/controllers/uploads.js]: Please not hardcode root url, or put in to constant if needed

File: server/controllers/uploads.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+router.post('/resource', function(req, res) {
+    req.pipe(req.busboy);
+    req.busboy.on('file', function(fieldname, file, filename) {
+        var fstream = fs.createWriteStream('./server/uploads/resources/' + filename); 
+        file.pipe(fstream);
+        fstream.on('close', function () {
+            res.json({
+                "uploaded": 1,
+                "fileName": filename,
+                "url": "http://localhost:8888/uploads/resources/"+filename
```
Comment (okrasn) [inline server/controllers/uploads.js]: The same as comment before

**Comment by drserg** [comment]
im trusting myself xD 
(joke)

**Pull request merged**